### ISSUE TITLE
Build error: identifier "make_Edge" is undefined

### ISSUE THREAD

**Issue by markerdmann** [mimblewimble/grin-miner#198]
Any idea what this could be?

Some info about my system:
Fedora 29
GTX 1080 Ti
CUDA 10.1.105
nvidia driver 418.43

Full stderr output:

```
--- stderr
Error copying file (if different) from "/home/sendrecv/apps/grin-miner/target/debug/build/cuckoo_miner-dd151cbcf863f6d6/out/build/CMakeFiles/cuckatoo_mean_cuda_19.dir/cuckoo/src/cuckatoo/cuckatoo_mean_cuda_19_generated_mean.cu.o.depend.tmp" to "/home/sendrecv/apps/grin-miner/target/debug/build/cuckoo_miner-dd151cbcf863f6d6/out/build/CMakeFiles/cuckatoo_mean_cuda_19.dir/cuckoo/src/cuckatoo/cuckatoo_mean_cuda_19_generated_mean.cu.o.depend".
CMake Error at cuckatoo_mean_cuda_19_generated_mean.cu.o.Debug.cmake:245 (message):
  Error generating
  /home/sendrecv/apps/grin-miner/target/debug/build/cuckoo_miner-dd151cbcf863f6d6/out/build/CMakeFiles/cuckatoo_mean_cuda_19.dir/cuckoo/src/cuckatoo/./cuckatoo_mean_cuda_19_generated_mean.cu.o


gmake[2]: *** [CMakeFiles/cuckatoo_mean_cuda_19.dir/build.make:264: CMakeFiles/cuckatoo_mean_cuda_19.dir/cuckoo/src/cuckatoo/cuckatoo_mean_cuda_19_generated_mean.cu.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:442: CMakeFiles/cuckatoo_mean_cuda_19.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
/home/sendrecv/apps/grin-miner/cuckoo-miner/src/cuckoo_sys/plugins/cuckoo/src/cuckatoo/mean.cu(111): error: identifier "make_Edge" is undefined
          detected during instantiation of "void SeedA<maxOut,EdgeOut>(const siphash_keys &, ulonglong4 *, u32 *) [with maxOut=544768, EdgeOut=u32]" 
(562): here

/home/sendrecv/apps/grin-miner/cuckoo-miner/src/cuckoo_sys/plugins/cuckoo/src/cuckatoo/mean.cu(129): error: identifier "make_Edge" is undefined
          detected during instantiation of "void SeedA<maxOut,EdgeOut>(const siphash_keys &, ulonglong4 *, u32 *) [with maxOut=544768, EdgeOut=u32]" 
(562): here

/home/sendrecv/apps/grin-miner/cuckoo-miner/src/cuckoo_sys/plugins/cuckoo/src/cuckatoo/mean.cu(203): error: identifier "make_Edge" is undefined
          detected during instantiation of "void SeedB<maxOut,EdgeOut>(const siphash_keys &, const EdgeOut *, ulonglong4 *, const u32 *, u32 *) [with maxOut=544768, EdgeOut=u32]" 
(578): here

3 errors detected in the compilation of "/tmp/tmpxft_00002428_00000000-14_mean.compute_75.cpp1.ii".
/home/sendrecv/apps/grin-miner/cuckoo-miner/src/cuckoo_sys/plugins/cuckoo/src/cuckatoo/mean.cu(111): error: identifier "make_Edge" is undefined
          detected during instantiation of "void SeedA<maxOut,EdgeOut>(const siphash_keys &, ulonglong4 *, u32 *) [with maxOut=512, EdgeOut=u32]" 
(562): here

/home/sendrecv/apps/grin-miner/cuckoo-miner/src/cuckoo_sys/plugins/cuckoo/src/cuckatoo/mean.cu(129): error: identifier "make_Edge" is undefined
          detected during instantiation of "void SeedA<maxOut,EdgeOut>(const siphash_keys &, ulonglong4 *, u32 *) [with maxOut=512, EdgeOut=u32]" 
(562): here

/home/sendrecv/apps/grin-miner/cuckoo-miner/src/cuckoo_sys/plugins/cuckoo/src/cuckatoo/mean.cu(203): error: identifier "make_Edge" is undefined
          detected during instantiation of "void SeedB<maxOut,EdgeOut>(const siphash_keys &, const EdgeOut *, ulonglong4 *, const u32 *, u32 *) [with maxOut=512, EdgeOut=u32]" 
(578): here

CMake Error at cuckatoo_mean_cuda_rtx_31_generated_mean.cu.o.Debug.cmake:279 (message):
  Error generating file
  /home/sendrecv/apps/grin-miner/target/debug/build/cuckoo_miner-dd151cbcf863f6d6/out/build/CMakeFiles/cuckatoo_mean_cuda_rtx_31.dir/cuckoo/src/cuckatoo/./cuckatoo_mean_cuda_rtx_31_generated_mean.cu.o


/home/sendrecv/apps/grin-miner/cuckoo-miner/src/cuckoo_sys/plugins/cuckoo/src/cuckatoo/mean.cu(111): error: identifier "make_Edge" is undefined
          detected during instantiation of "void SeedA<maxOut,EdgeOut>(const siphash_keys &, ulonglong4 *, u32 *) [with maxOut=552960, EdgeOut=u32]" 
(562): here

/home/sendrecv/apps/grin-miner/cuckoo-miner/src/cuckoo_sys/plugins/cuckoo/src/cuckatoo/mean.cu(111): error: identifier "make_Edge" is undefined
          detected during instantiation of "void SeedA<maxOut,EdgeOut>(const siphash_keys &, ulonglong4 *, u32 *) [with maxOut=512, EdgeOut=u32]" 
(562): here

/home/sendrecv/apps/grin-miner/cuckoo-miner/src/cuckoo_sys/plugins/cuckoo/src/cuckatoo/mean.cu(129): error: identifier "make_Edge" is undefined
          detected during instantiation of "void SeedA<maxOut,EdgeOut>(const siphash_keys &, ulonglong4 *, u32 *) [with maxOut=552960, EdgeOut=u32]" 
(562): here

/home/sendrecv/apps/grin-miner/cuckoo-miner/src/cuckoo_sys/plugins/cuckoo/src/cuckatoo/mean.cu(129): error: identifier "make_Edge" is undefined
          detected during instantiation of "void SeedA<maxOut,EdgeOut>(const siphash_keys &, ulonglong4 *, u32 *) [with maxOut=512, EdgeOut=u32]" 
(562): here

gmake[2]: *** [CMakeFiles/cuckatoo_mean_cuda_rtx_31.dir/build.make:271: CMakeFiles/cuckatoo_mean_cuda_rtx_31.dir/cuckoo/src/cuckatoo/cuckatoo_mean_cuda_rtx_31_generated_mean.cu.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
/home/sendrecv/apps/grin-miner/cuckoo-miner/src/cuckoo_sys/plugins/cuckoo/src/cuckatoo/mean.cu(203): error: identifier "make_Edge" is undefined
          detected during instantiation of "void SeedB<maxOut,EdgeOut>(const siphash_keys &, const EdgeOut *, ulonglong4 *, const u32 *, u32 *) [with maxOut=552960, EdgeOut=u32]" 
(578): here

/home/sendrecv/apps/grin-miner/cuckoo-miner/src/cuckoo_sys/plugins/cuckoo/src/cuckatoo/mean.cu(203): error: identifier "make_Edge" is undefined
          detected during instantiation of "void SeedB<maxOut,EdgeOut>(const siphash_keys &, const EdgeOut *, ulonglong4 *, const u32 *, u32 *) [with maxOut=512, EdgeOut=u32]" 
(578): here

3 errors detected in the compilation of "/tmp/tmpxft_0000241e_00000000-14_mean.compute_75.cpp1.ii".
3 errors detected in the compilation of "/tmp/tmpxft_00002423_00000000-14_mean.compute_75.cpp1.ii".
3 errors detected in the compilation of "/tmp/tmpxft_00002421_00000000-14_mean.compute_75.cpp1.ii".
CMake Error at cuckatoo_mean_cuda_19_generated_mean.cu.o.Debug.cmake:279 (message):
  Error generating file
  /home/sendrecv/apps/grin-miner/target/debug/build/cuckoo_miner-dd151cbcf863f6d6/out/build/CMakeFiles/cuckatoo_mean_cuda_19.dir/cuckoo/src/cuckatoo/./cuckatoo_mean_cuda_19_generated_mean.cu.o


CMake Error at cuckatoo_mean_cuda_19_generated_mean.cu.o.Debug.cmake:279 (message):
  Error generating file
  /home/sendrecv/apps/grin-miner/target/debug/build/cuckoo_miner-dd151cbcf863f6d6/out/build/CMakeFiles/cuckatoo_mean_cuda_19.dir/cuckoo/src/cuckatoo/./cuckatoo_mean_cuda_19_generated_mean.cu.o


gmake[2]: *** [CMakeFiles/cuckatoo_mean_cuda_rtx_31.dir/build.make:264: CMakeFiles/cuckatoo_mean_cuda_19.dir/cuckoo/src/cuckatoo/cuckatoo_mean_cuda_19_generated_mean.cu.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:627: CMakeFiles/cuckatoo_mean_cuda_rtx_31.dir/all] Error 2
CMake Error at cuckatoo_mean_cuda_gtx_31_generated_mean.cu.o.Debug.cmake:279 (message):
  Error generating file
  /home/sendrecv/apps/grin-miner/target/debug/build/cuckoo_miner-dd151cbcf863f6d6/out/build/CMakeFiles/cuckatoo_mean_cuda_gtx_31.dir/cuckoo/src/cuckatoo/./cuckatoo_mean_cuda_gtx_31_generated_mean.cu.o


gmake[2]: *** [CMakeFiles/cuckatoo_mean_cuda_gtx_31.dir/build.make:264: CMakeFiles/cuckatoo_mean_cuda_19.dir/cuckoo/src/cuckatoo/cuckatoo_mean_cuda_19_generated_mean.cu.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[2]: *** [CMakeFiles/cuckatoo_mean_cuda_gtx_31.dir/build.make:271: CMakeFiles/cuckatoo_mean_cuda_gtx_31.dir/cuckoo/src/cuckatoo/cuckatoo_mean_cuda_gtx_31_generated_mean.cu.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:590: CMakeFiles/cuckatoo_mean_cuda_gtx_31.dir/all] Error 2
gmake: *** [Makefile:84: all] Error 2
thread 'main' panicked at '
command did not execute successfully, got: exit code: 2

build script failed, must exit now', /home/sendrecv/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.34/src/lib.rs:776:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
   1: std::sys_common::backtrace::print
   2: std::panicking::default_hook::{{closure}}
   3: std::panicking::default_hook
   4: std::panicking::rust_panic_with_hook
   5: std::panicking::continue_panic_fmt
   6: std::panicking::begin_panic_fmt
   7: cmake::fail
             at /home/sendrecv/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.34/src/lib.rs:776
   8: cmake::run
             at /home/sendrecv/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.34/src/lib.rs:757
   9: cmake::Config::build
             at /home/sendrecv/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.34/src/lib.rs:669
  10: build_script_build::main
             at cuckoo-miner/src/build.rs:75
  11: std::rt::lang_start::{{closure}}
             at /builddir/build/BUILD/rustc-1.32.0-src/src/libstd/rt.rs:74
  12: std::panicking::try::do_call
  13: __rust_maybe_catch_panic
  14: std::rt::lang_start_internal
  15: std::rt::lang_start
             at /builddir/build/BUILD/rustc-1.32.0-src/src/libstd/rt.rs:74
  16: main
  17: __libc_start_main
  18: _start

warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by kylemiller3** [comment]
Same issue on Ubuntu 18.04
nvcc V10.1.105
driver 418.56

**Comment by tromp** [comment]
Should be fixed by tromp/cuckoo@93975e9
courtesy of https://github.com/agusdallalba

To fix this before the next grin-miner update, you can run
git submodule update --remote --recursive
and rebuild grin-miner
